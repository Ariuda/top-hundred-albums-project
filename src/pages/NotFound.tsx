import React from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';

class NotFound extends React.Component<RouteComponentProps> {
    state = {
        timeoutId: 0
    }
    componentDidMount() {
        this.setState({ timeoutId: setTimeout(() => {
            this.props.history.push('/');
        }, 5000)});
    }
    componentWillUnmount() {
        clearTimeout(this.state.timeoutId);
    }

    render() {
        
        return (
            <div className="not-found">
                <h1>Top 100 albums in the iTunes Store</h1>
                <div className="not-found-card">
                    <p>Sorry, we couldn't find the page you are looking for</p>
                    <p>You'll be redirected to our home page in 5 seconds.</p>
                    <p>Alternatively, <Link to="/">click here</Link> to go back.</p>
                </div>
            </div>
        )
    }

};

export default NotFound;