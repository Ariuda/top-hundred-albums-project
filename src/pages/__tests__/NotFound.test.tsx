import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { NotFound } from '../NotFound';

const Props = {
    history: {} as any,
    location: {} as any,
    match: {} as any
};

it('should render the not found component correctly', () => {
    const wrapper = shallow(<NotFound {...Props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});