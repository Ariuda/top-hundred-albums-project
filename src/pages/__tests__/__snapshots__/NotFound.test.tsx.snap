// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the not found component correctly 1`] = `
<div
  className="not-found"
>
  <h1>
    Top 100 albums in the iTunes Store
  </h1>
  <div
    className="not-found-card"
  >
    <p>
      Sorry, we couldn't find the page you are looking for
    </p>
    <p>
      You'll be redirected to our home page in 5 seconds.
    </p>
    <p>
      Alternatively, 
      <Link
        to="/"
      >
        click here
      </Link>
       to go back.
    </p>
  </div>
</div>
`;
